utils/pckgen

```diff
@@ -0,0 +1,559 @@
+        fi
+
+        CREATION_DATE=$(date --iso-8601=seconds)
+        CREATION_TIMESTAMP=$(date +%s)
+        echo "CREATION_DATE=$CREATION_DATE" > "$VERSION_FILE"
+        echo "CREATION_TIMESTAMP=$CREATION_TIMESTAMP" >> "$VERSION_FILE"
+        echo "CAS_VERSION_MAIN=$CAS_VERSION_MAIN" >> "$VERSION_FILE"
+        echo "CAS_VERSION_MAJOR=$CAS_VERSION_MAJOR" >> "$VERSION_FILE"
+        echo "CAS_VERSION_MINOR=$CAS_VERSION_MINOR" >> "$VERSION_FILE"
+        echo "CAS_VERSION_BUILD=$CAS_VERSION_BUILD" >> "$VERSION_FILE"
+        echo "LAST_COMMIT_HASH=$LAST_COMMIT_HASH" >> "$VERSION_FILE"
+        echo "LAST_ABBR_COMMIT_HASH=$LAST_ABBR_COMMIT_HASH" >> "$VERSION_FILE"
+        echo "LAST_COMMIT_DATE=$LAST_COMMIT_DATE" >> "$VERSION_FILE"
+        echo "LAST_COMMIT_TIMESTAMP=$LAST_COMMIT_TIMESTAMP" >> "$VERSION_FILE"
+    fi
+}
+
+archive_prepare() {
+    echo "--- Preparing sources to archive"
+    if [ $ARCHIVE_PREPARED ] && [ ! $FORCE ]; then
+        echo "---   Sources already prepared (use --force to overwrite)"
+    else
+        echo "--- Making copy to work on"
+        local TEMP_SOURCES_DIR="$TEMP_DIR/$CAS_FILENAME"
+        rm -rf "$TEMP_SOURCES_DIR"
+        mkdir -p "$TEMP_SOURCES_DIR"
+        cp -fat "$TEMP_SOURCES_DIR" "$SOURCES_DIR/."
+
+        if [ -d "$TEMP_SOURCES_DIR/.git" ]; then
+            echo "--- Updating submodules"
+            for SUBMOD in ${SUBMODULES[@]}; do
+                if [ ! -f "$TEMP_SOURCES_DIR/$SUBMOD/.git" ]; then
+                    local SUBMODULES_UPDATE=1
+                fi
+            done
+            if [ ! $SUBMODULES_UPDATE ] && [ ! $FORCE ]; then
+                echo "---   Submodules already initialized (use --force to update)"
+            else
+                if [ $NO_NET ]; then
+                    info "---   Submodules not updated - '--no-download' option is set"
+                else
+                    git -C "$TEMP_SOURCES_DIR" submodule update --init
+                    if [ $? -ne 0 ]; then
+                        error "Couldn't update submodules"
+                    fi
+                fi
+            fi
+            for SUBMOD in ${SUBMODULES[@]}; do
+                if [ ! -f "$TEMP_SOURCES_DIR/$SUBMOD/.git" ] && [ ! $NO_NET ]; then
+                    error "Couldn't update submodules"
+                fi
+            done
+        fi
+
+        echo "--- Removing git tree"
+        find "$TEMP_SOURCES_DIR" -name ".git*" -print0 | xargs -0 rm -rf
+        find "$TEMP_SOURCES_DIR" -name .pep8speaks.yml -type f -print0 | xargs -0 rm -f
+        if ! find "$TEMP_SOURCES_DIR" -name ".git*" -exec false {} +; then
+            error "Couldn't remove git tree"
+        fi
+
+        ARCHIVE_PREPARED=1
+    fi
+}
+
+generate_tar() {
+    echo "--- Creating tar archive from current sources"
+    if [ -f "$OUTPUT_DIR/$SOURCES_TAR_NAME" ] && [ ! $FORCE ]; then
+        echo "---   Tarball already present in output folder (use --force to overwrite)"
+    else
```

Comment:
Typically we don't care. When we run `make`, the new binaries will just replace the old ones. Why should we care in case of tar package?

Reply:
This check is here to prevent building tar package if it was already built by e.g. 'generate_rpm()'. But I can see it now being very unintuitive. I'll find a different solution.